sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -106,6 +108,22 @@ object TypeCoercion {
     case (t1, t2) => findTypeForComplex(t1, t2, findTightestCommonType)
   }
 
+  /**
+   * Finds a wider decimal type between the two supplied decimal types without
+   * any loss of precision.
+   */
+  def findWiderDecimalType(d1: DecimalType, d2: DecimalType): Option[DecimalType] = {
+    val scale = max(d1.scale, d2.scale)
+    val range = max(d1.precision - d1.scale, d2.precision - d2.scale)
+
+    // Check the resultant decimal type does not exceed the allowable limits.
+    if (range + scale <= DecimalType.MAX_PRECISION && scale <= DecimalType.MAX_SCALE ) {
```

Comment:
super nit: unnecessary space `DecimalType.MAX_SCALE )`

Reply:
Is `scale <= DecimalType.MAX_SCALE` necessary? If `d1` and `d2` are valid and their `scale` didn't reach `DecimalType.MAX_SCALE`, so the same condition is kept for `max(d1.scale, d2.scale)`

Reply:
@MaxGekk You are right. I was not sure if we could come here with a invalid decimal i.e scale > MAX_SCALE.. Basically i looked at the bound method which does a min(scale, MAX_SCALE) and modelled it like that here to be defensive.